client/components/domain-diagnostic/index.jsx

```diff
@@ -0,0 +1,42 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+import Notice from 'components/notice';
+import NoticeAction from 'components/notice/notice-action';
+
+export default React.createClass( {
+
+	displayName: 'DomainDiagnostic',
+
+	propTypes: {
+		domain: React.PropTypes.string.isRequired,
+	},
+	getInitialState: function() {
+		return {
+			domainDidFail: false
+		};
+	},
+
+	componentDidMount() {
+		const self = this;
+		const imageChecker = new Image();
+		imageChecker.onerror = function() {
+			self.setState( { domainDidFail: true } );
+		}
+		imageChecker.src = 'http://' + this.props.domain + '/w32.gif';
```

Comment:
> This looks super fragile, what happens if the image is removed? 

It has been there for 10 years, but it will show errors. We can pick another file though

> Also, if we do this it should be https - that url is just going to redirect to https after all and we serve calypso over https.

Yes, we serve wordpress.com domain over https. All the the other domains we do not! I am checking domains like piszek.com